services/refCompiler.js

```diff
@@ -0,0 +1,154 @@
+var fs = require('fs');
+var path = require('path');
+var specsConfig;
+var thisLogger;
+exports.init = function (logger, config)  {
+    specsConfig = config.get('refCompiler');
+    thisLogger = logger;
+};
+
+exports.compileSpecs = function () {
+    for (var key in specsConfig) {
+        if (/.+\.(yaml)$/i.test(specsConfig[key].baseSpecFile)) {
+            compileYAMLReferences(specsConfig[key].baseSpecFile, specsConfig[key].refDirs);
+        } else if (/.+\.(json)$/i.test(specsConfig[key].baseSpecFile)) {
+            compileJSONReferences(specsConfig[key].baseSpecFile, specsConfig[key].refDirs);
+        } else {
+            thisLogger.warn('Unknown file extension for base spec file %s', specsConfig[key].baseSpecFile);
+        }
+    }
+};
+
+/**
+ * @param  {string} baseSpecFile - file name of the base spec
+ *  e.g. 'app-api-mysf-v2.yaml'
+ * @param  {string[]} refDirs - list of directories with references to include
+ *  e.g. ['v2/public', 'v2/private/mysf']
+ */
+function compileYAMLReferences(baseSpecFile, refDirs) {
+    var swaggerDir = path.resolve(global.__appDir, 'swagger');
```

Comment:
This I think may need to be changed to use the 'context' value of the swagger service config. Either that or follow the same isBlueOakProject() logic specified in the swagger service. Please confirm which of these if any solutions are appropriate.